### DISCUSSION TITLE
Alerting when calibration results in a significant slope/intercept shift

### DISCUSSION THREAD

**Discussion by old-square-eyes** [NightscoutFoundation/xDrip#4095] [Ideas]
Currently, a calibration can result in a significant slope shift. Sometimes this might be desired (i.e. bringing in a slope to where it should be, perhaps after a bump on a sensor).

Twice recently I have calibrated, and it has had an undesirable effect on my slope. I haven't noticed, and it has resulted in xDrip interpreting that I am 1-2 mmol/L higher than I actually was. This to me seems quite risky.

I wonder if there could be a toast / alert to warn that this has happened - with a sound?

"Warning! Calibration slope has changed significantly. Do you wish to accept?"

This is so the user can review the chart and potentially reconsider the accuracy of the calibration and otherwise act on it. The sound is since auto calibration is not accepted until 15 minutes after the blood test and a toast may be missed on its own.

![screenshot_20180226-084559](https://user-images.githubusercontent.com/25760491/36648824-0fc581ce-1a90-11e8-96e7-a19811362986.png)
![screenshot_20180226-084619](https://user-images.githubusercontent.com/25760491/36648825-0ff5ed64-1a90-11e8-8d29-000173125640.png)

**Comment by Navid200** [comment]
@old-square-eyes 
When you calibrate, either the slope will be affected or the intercept or both.

I understand you like to have a warning when the slope is changed too much.  
But, someone else may also like to get a warning when the intercept has changed too much.
I think the solution is to just check both intercept and slope after every calibration to be safe.  
If you make that a habit, you will not need a warning.  Do you agree?

**Comment by old-square-eyes** [comment]
Yes, I should have said slope and intercept. Though many people don't understand it, so simple language is probably best. "Calibration has changed significantly, are you sure you wish to proceed. Please consider testing again, and only calibrate with low noise and stable blood sugar.", or something like that. There are definately times when a big shift is what you want, so it's shouldn't be a brick wall.

**Comment by Navid200** [comment]
Considering you have been a part of the community longer than me, I am just making suggestions for your consideration.
In my opinion, we diabetics need alarms to tell us that our glucose is high or low.

But, I think we can survive if we don't get an alarm when the calibration slope or intercept has changed.  That happens when we calibrate and we should always check the result of the calibration.

So, my question for you is this.  Are you sure you want to ask for a new alarm?  Considering that it has to be developed and there are so many other feature requests already open.  I hope you agree to allow this one to go.
Otherwise, would you be willing to let me label this as postponed?  So, if we get to a point when other features are addressed, we can reconsider and reopen it.

**Comment by old-square-eyes** [comment]
Couple of things. Agree that more alarms is potentially annoying for users, and extra dev work. Secondly, the number of people using raw data and calibrating on xDrip is shrinking due to the new Dexcom transmitters (not sure what other brands this feature would service). It's a very old ticket and I use Native myself now. Though, logs and UX in xDrip does not quite gel that well with Native users - impossible slope/initial calibration experience/errors etc, which are irellivant for Native users - but that's another Issue).

This ticket was from when I was using auto calibration feature in xDrip. By and large it does a good job of automatically accepting. But as described, there were times when unwittingly (and others automatically) a big, and incorrect, calibration shift was entered/accepted. The audio alarm suggested was for times when auto calibrate accepted something that resulted in the shift... the toast was for somone manually entering and accepting a calibration, and just serves as a gentle reminder to check thier calibration graph, and if the new plot is way off, to consider the reasons for that, and possibly remove and retry the calibration.

**Comment by old-square-eyes** [comment]
From a safety perspective I only calibrate once a day, even on Raw. So a big shift, undetected, can result in a day of wrong readings. High or low, it's not great.

**Comment by Navid200** [comment]
There has been no development on this.
I would like to either close the issue or convert it to a discussion.

Please tell me if you have a preference.
Thanks

**Comment by Navid200** [comment]
I am converting this to a discussion.  It will still be found through search.  A developer could still see it and decide to implement it.  

I personally hardly ever calibrate now that I use G7.  I hardly ever use my blood glucose meter.  Only if my symptoms disagree with what the CGM shows, do I use my blood glucose meter.  
Calibrations submitted to a G7 affects G7.  But, there is no local slope or intercept anyway in that case.

I'm not suggesting this is an invalid request.  I am just making an observation in agreement with you that less people calibrate nowadays compared to when this request was first submitted.

Thanks
